interactive.c

```diff
@@ -116,23 +155,36 @@ void interactiveShowData(void) {
 
     next_update = now + MODES_INTERACTIVE_REFRESH_TIME;
 
-    mvprintw(0, 0, " Hex    Mode  Sqwk  Flight   Alt    Spd  Hdg    Lat      Long   RSSI  Msgs  Ti");
-    mvhline(1, 0, ACS_HLINE, 80);
+    if (Modes.interactive_show_distance) {
+        mvprintw(1, 0, " Hex    Mode  Sqwk  Flight   Alt    Spd  Hdg  Dist(%s) Bearing  RSSI  Msgs  Ti",
+            distance_units_suffix);
+    } else {
+        mvprintw(1, 0, " Hex    Mode  Sqwk  Flight   Alt    Spd  Hdg    Lat      Long   RSSI  Msgs  Ti");
+    }
+
+    mvhline(2, 0, ACS_HLINE, 80);
 
     progress = spinner[(now/1000)%4];
     mvaddch(0, 79, progress);
 
     int rows = getmaxy(stdscr);
-    int row = 2;
+    int row = 3;
+    int rowMaxd = 0;
+    int rowMaxRSSI = 0;
+    int rowMinRSSI = 0;
 
-    while (a && row < rows) {
-        if (a->reliable && (now - a->seen) < Modes.interactive_display_ttl) {
+    while (a) {
+
+        if (a->reliable && (now - a->seen) < Modes.interactive_display_ttl
+            && (Modes.interactive_callsign_filter == NULL
+            || regexec(&callsign_filter_regex, a->callsign, 0, NULL, 0) == 0)) {
```

Comment:
Most of the time, a->callsign does not change. Can you cache the result of regexec here so it's not run on every refresh? (You'll need a "callsign changed" bit somewhere in aircraft state too)

Reply:
Good idea.